internal/graph/graph.go

```diff
@@ -122,10 +149,37 @@ func BuildConnectedObjectGraph(typesystem *typesystem.TypeSystem) *ConnectedObje
 //  1. If parent's types includes `folder` type, and `folder` contains `viewer` relation then this is exactly a ttu rewrite and....?
 //  2. Otherwise, suppose the types contains `objectType` which has a relation `viewer`, then recurse on `objectType#viewer, folder#viewer`
 func (g *ConnectedObjectGraph) RelationshipIngresses(target *openfgapb.RelationReference, source *openfgapb.RelationReference) ([]*RelationshipIngress, error) {
-	return g.findIngresses(target, source, map[string]struct{}{})
+	return g.findIngresses(target, source, map[string]struct{}{}, resolveAllIngresses)
 }
 
-func (g *ConnectedObjectGraph) findIngresses(target *openfgapb.RelationReference, source *openfgapb.RelationReference, visited map[string]struct{}) ([]*RelationshipIngress, error) {
+// PrunedRelationshipIngresses compute the minimum incoming endges (ingresses) that are possible between the target and the source.
+//
+// PrunedRelationshipIngresses is primarily used as the driver behind ingresses that work for models including intersection and exclusion.
+// If the ingresses from the source to the target pass through a relationship involving intersection or exclusion (directly or indirectly),
+// then the PrunedRelationshipIngresses will just return the first-most ingress involved in the rewrite.
```

Comment:
Consider the following model:
```
type user

type group
  relations
    define member: [user]

type document
  relations
    define editor: [user]
    define viewer: [user, group#member] and editor
```

If we consider the _unpruned_ ingresses from `user` to `document#viewer`, then there are 3 ingresses. These include [`group#member`, `document#viewer`, `document#editor`]. In order to evaluate `document#viewer` relationships you'd have to evaluate all of these ingresses.

If we consider _pruned_ ingreses from `user` to `document#viewer`, then there are 2 ingresses. These include [`group#member`, `document#viewer`], because the `document#editor` ingress drops out as an optimization since it is contained under the intersection.

This function returns a slice because there can be more than 1 ingresses, but the pruned ingress list removes ingresses contained under an intersection or exclusion.

Reply:
Ok so `PrunedRelationshipIngresses` is like a superset of `RelationshipIngresses` in that depending on the model, it can return one or more elements?

If so I suggest updating the documentation to:

```
// If the ingresses from the source to the target pass through a relationship involving intersection or exclusion (directly or indirectly),
// then the PrunedRelationshipIngresses will just return the first-most ingress involved in the rewrite. Otherwise, it will return all of them.
```

Reply:
`PrunedRelationshipIngresses` is a subset of RelationshipIngresses. The number of ingresses returned from `PrunedRelationhipIngresses` would always be less than that returned by `RelationshipIngresses`, because the later returns all ingresses, whereas the pruned variant only returns the first ingress involved in an intersection or exclusion rewrite operator.